src/orchard/java.clj

```diff
@@ -141,16 +116,15 @@
   (if (>= misc/java-api-version 9)
     (do (require '[orchard.java.parser :as src])
         (resolve 'src/source-info))
-    (if jdk-tools
-      (do (require '[orchard.java.legacy-parser :as src])
-          (resolve 'src/source-info))
-      (constantly nil))))
+    (do
+      (assert jdk-tools)
```

Comment:
Probably the `assert` is dangerous here, we cannot be sure of whether Orchard is being used with a plugin that will  place the tools on the classpath.

Relatedly, in cider.el it can be hard to conditionally inject the tools plugin depending on the detected JDK. 

For these reasons it seems good to move the tools logic to enrich-classpath which would do the JDK detection.

Reply:
Yeah, that makes perfect sense, but I guess it will need at least clojure cli support first.

Reply:
Btw, do we need an `assert` here - it might a bit frustrating for users that Orchard will just blow up there. Perhaps the conditional approach that does nothing if `jdk-tools` is missing is better?

Reply:
Yeah that's what I meant w/ `Probably the assert is dangerous here`. Probably when I wrote this code a few weeks ago I meant the assert as a "fix this before merging it" thing i.e. a glorified TODO 😄

btw, yesterday I progressed a little https://github.com/clojure-emacs/enrich-classpath/commit/4cb44de5502494b9806aa570cc410ed87ffbcf6b